src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -1310,16 +1311,26 @@ public function searchProductsAction(Request $request): Response
     {
         try {
             $searchPhrase = $request->query->get('search_phrase');
+            $currencyId = $request->query->get('search_currency');
+            $currency = null;
+
+            /** @var CurrencyDataProviderInterface $currencyDataProvider */
+            $currencyDataProvider = $this->container->get('prestashop.adapter.data_provider.currency');
```

Comment:
Ok it's lame that `CurrencyDataProviderInterface` returns legacy `Currency` objects.. So maybe `SearchProducts` should only contain the currency iso code directly then in this controller you can use `Currency::getIsoCodeById`

Reply:
:ballot_box_with_check: